regression_detector.py

```diff
@@ -0,0 +1,69 @@
+"""
+The regression detector of TorchBench Userbenchmark.
+"""
+import json
+import argparse
+import importlib
+from dataclasses import dataclass, asdict
+import os
+import yaml
+from pathlib import Path
+import time
+from datetime import datetime
+from typing import Optional
+from userbenchmark.utils import PLATFORMS, USERBENCHMARK_OUTPUT_PREFIX, REPO_PATH, \
+                                TorchBenchABTestResult
+
+
+def call_userbenchmark_detector(detector, start_file: str, end_file: str) -> Optional[TorchBenchABTestResult]:
+    return detector(start_file, end_file)
+
+def get_default_output_path(bm_name: str) -> str:
+    output_path = os.path.join(REPO_PATH, USERBENCHMARK_OUTPUT_PREFIX, bm_name)
+    fname = "regression-{}.yaml".format(datetime.fromtimestamp(time.time()).strftime("%Y%m%d%H%M%S"))
+    return os.path.join(output_path, fname)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    # Local metrics file comparison
+    parser.add_argument("--control", default=None, help="The control group metrics file for comparison.")
+    parser.add_argument("--treatment", default=None, help="The treatment metrics file for comparison.")
+
+    # S3 metrics file comparison
+    parser.add_argument("--name", help="Name of the userbenchmark to detect regression.")
+    parser.add_argument("--platform", choices=PLATFORMS, default=None, help="The name of platform of the regression.")
+    parser.add_argument("--start-date", default=None, help="The start date to detect regression.")
+    parser.add_argument("--end-date", default=None, help="The latest date to detect regression.")
```

Comment:
how are these to be used? this looks like remnant code from the bisector.

Reply:
Later we will add ability to download metrics from Amazon S3 for this script. These arguments will be used to identify the S3 bucket prefix:
```
ossci-metrics/torchbench-userbenchmark/<userbenchmark-name>/<platform>/<date>
```

By specifying userbenchmark-name, platform, and start/end dates, this script can download the metrics to a directory, compare them locally, and upload the result back to S3.